lamindb/core/storage/objects.py

```diff
@@ -1,62 +1,95 @@
 from __future__ import annotations
 
 from pathlib import PurePosixPath
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, TypeAlias, TypeVar
 
 from anndata import AnnData
 from pandas import DataFrame
 
 if TYPE_CHECKING:
     from lamindb_setup.core.types import UPathStr
 
+SpatialData = TypeVar("SpatialData")
+MuData = TypeVar("MuData")
 
-def _mudata_is_installed():
-    try:
-        import mudata  # noqa: F401c
-    except ImportError:
-        return False
-    return True
+SupportedDataTypes: TypeAlias = AnnData | DataFrame | MuData | SpatialData
 
 
-def infer_suffix(dmem, adata_format: str | None = None):
+def is_package_installed(package_name):
+    import importlib.util
+
+    spec = importlib.util.find_spec(package_name)
+    return spec is not None
+
+
+def infer_suffix(dmem: SupportedDataTypes, format: str | None = None):
     """Infer LaminDB storage file suffix from a data object."""
     if isinstance(dmem, AnnData):
-        if adata_format is not None:
-            if adata_format not in {"h5ad", "zarr", "anndata.zarr"}:
+        if format is not None:
+            if format not in {"h5ad", "zarr", "anndata.zarr"}:
                 raise ValueError(
                     "Error when specifying AnnData storage format, it should be"
-                    f" 'h5ad', 'zarr', not '{adata_format}'. Check 'format'"
+                    f" 'h5ad', 'zarr', not '{format}'. Check 'format'"
                     " or the suffix of 'key'."
                 )
-            return "." + adata_format
+            return "." + format
         return ".h5ad"
-    elif isinstance(dmem, DataFrame):
+
+    if isinstance(dmem, DataFrame):
         return ".parquet"
-    else:
-        if _mudata_is_installed():
-            from mudata import MuData
 
-            if isinstance(dmem, MuData):
-                return ".h5mu"
+    if is_package_installed("mudata"):
+        from mudata import MuData
+
+        if isinstance(dmem, MuData):
+            return ".h5mu"
+
+    if is_package_installed("spatialdata"):
+        from spatialdata import SpatialData
+
+        if isinstance(dmem, SpatialData):
+            if format is not None:
+                if format not in {"spatialdata.zarr"}:
```

Comment:
Why is there not `zarr` here? I thought I had it before and it would be consistent with the error message and the behavior for anndata just above.

Reply:
I need to dig through changes again; will investigate whether this was conscious or not.

Please no changes currently to main however; I spent >14 hours already on the re-structure and am pretty close to merging it.

Then one can polish again without causing conflicts.